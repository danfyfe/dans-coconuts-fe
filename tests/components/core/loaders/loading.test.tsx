import Loading from "@/components/core/loaders/loading";
import { render, RenderResult } from "@testing-library/react";

describe('Loading component', () => {
  let component: RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>;
  let element: HTMLElement;
  const className = 'mb-2';
  const coconutImageSrc = encodeURIComponent('/images/coconut.png');

  beforeEach(() => {
    component = render(
      <Loading className={className} />
    );
    element = component.getByTestId('loading-component')
  });

  it('renders', () => {
    expect(component).toBeDefined();
    expect(element).toBeDefined();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders a coconut image', () => {
    const image = element.getElementsByTagName('IMG')[0];
    const src = image.getAttribute('src');
    // NextJS Image component adds text to src, check for base inside
    expect(src).toContain(coconutImageSrc);
    expect(image).toBeDefined();
  });

  it('renders a span with loading text', () => {
    const span = component.getByText('...Loading...');
    expect(span).toBeDefined();
    expect(span.tagName).toBe('SPAN');
  });

  it('uses the passed in className', () => {
    expect(element.classList).toContain('mb-2');
  });
});
