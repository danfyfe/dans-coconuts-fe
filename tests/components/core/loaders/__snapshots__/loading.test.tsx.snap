// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
					relative w-full h-full m-2 flex flex-col
					justify-center items-center bg-sand_day
					z-[1000]
					mb-2
				"
        data-testid="loading-component"
      >
        <img
          alt="An image of a coconut - Loading"
          class="h-16 w-16 animate-[spin_1.5s_infinite]"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="100"
          src="/_next/image?url=%2Fimages%2Fcoconut.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2Fcoconut.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fcoconut.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
        <div
          class="overflow-hidden w-max mt-3"
        >
          <span
            class=" animate-left-to-right relative"
          >
            ...Loading...
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
					relative w-full h-full m-2 flex flex-col
					justify-center items-center bg-sand_day
					z-[1000]
					mb-2
				"
      data-testid="loading-component"
    >
      <img
        alt="An image of a coconut - Loading"
        class="h-16 w-16 animate-[spin_1.5s_infinite]"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="100"
        src="/_next/image?url=%2Fimages%2Fcoconut.png&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2Fcoconut.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fcoconut.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
      <div
        class="overflow-hidden w-max mt-3"
      >
        <span
          class=" animate-left-to-right relative"
        >
          ...Loading...
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
